### PR TITLE
Register the EXTERNAL_QUERY tvf

### PR DESCRIPTION
**Pull request by lukehalasy-anchorage**
Hello. I meant to open this PR into my personal fork, but I will leave it here in case anyone else is ever interested in this.
The dbt project (BigQuery based) that I use this language server on utilizes a lot of EXTERNAL_QUERY's. 
https://cloud.google.com/bigquery/docs/reference/standard-sql/federated_query_functions#external_query

The language server throws errors that say "Table-valued function not found: EXTERNAL_QUERY". 
This PR fixes the errors. 
If someone else wants to help me think through how to deliver this in a smarter/more thorough way I would be happy hear it. Thanks for this great plugin!

### CODE REVIEW

File: server/src/ZetaSqlApi.ts [hunk 1]
```diff
@@ -104,6 +104,8 @@ export class ZetaSqlApi {
         const featuresForVersion = await this.zetaSql.LanguageOptions.getLanguageFeaturesForVersion(LanguageVersion.VERSION_CURRENT);
         featuresForVersion.forEach(f => options.enableLanguageFeature(f));
         options.enableLanguageFeature(LanguageFeature.FEATURE_NAMED_ARGUMENTS);
+        options.enableLanguageFeature(LanguageFeature.FEATURE_CREATE_TABLE_FUNCTION);
+        options.enableLanguageFeature(LanguageFeature.FEATURE_CREATE_EXTERNAL_TABLE_WITH_CONNECTION);
```
Comment (lukehalasy-anchorage) [inline server/src/ZetaSqlApi.ts:108]: I don't think line 108 is needed

**Comment by pgrivachev** [comment]
Hello @lukehalasy-anchorage,

Thank you for this PR! I think you are on right way, but we should fill catalog with signature and return type. As I understand it takes 3 string arguments (3rd is optional). I will take a look on how to do it soon. 

It will be great if you show any simple example of usage `EXTERNAL_QUERY` function.

**Comment by lukehalasy-anchorage** [comment]
Also thanks for raising the issue here -> https://github.com/google/zetasql/issues/145

**Comment by lukehalasy-anchorage** [comment]
It looks like Google occasionally updates the ZetaSQL repo with internal changes (about once a month).
 I think that internally they already have this setup. The reason I think this is because the BigQuery IDE can report syntax errors within EXTERNAL_QUERY's.  It will likely just be a waiting game on Google.

It would be awesome to have auto-complete/error showing within the EXTERNAL_QUERY's but that would require keeping track of tables and columns within the CloudSQL database that the EXTERNAL_QUERY's connection is hooked up.

**Comment by pgrivachev** [comment]
@lukehalasy-anchorage thank you for this example! `signature` can help us specify types, but it is not a problem.
The problem now is that we probably can't declare tvf without explicit return value that should be something like
```
customTvf: [
        {
          namePath: ['external_query'],
          type: '...',
          signature: {
            argument: [...],

            returnType: {
              kind: SignatureArgumentKind.ARG_TYPE_RELATION,
              options: { relationInputSchema: {column:[]} },
            },
          },
```
But we don't know the resulting columns. Your solution doesn't have enough data for tvf declaration, you probably didn't see errors because ZetaSQL stopped working.

**Pull request closed without merging**